src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -104,6 +97,25 @@ class LedgerImpl(vm: VM, blockchainConfig: BlockchainConfig) extends Ledger with
     blockTxsExecResult
   }
 
+
+  private[ledger] def executePreparedTransactions(
+    signedTransactions: Seq[SignedTransaction], world: InMemoryWorldStateProxy,
+    blockHeader: BlockHeader, signedTransactionValidator: SignedTransactionValidator,
+    acumGas: BigInt = 0, acumReceipts: Seq[Receipt] = Nil, executed: Seq[SignedTransaction] = Nil)
+  : Either[TxsExecutionError, (BlockResult, Seq[SignedTransaction])] = {
+
+    val result = executeTransactions(signedTransactions, world, blockHeader, signedTransactionValidator, acumGas, acumReceipts)
+
+    result match {
+      case Left(TxsExecutionError(stx, Some(StateBeforeFailure(worldState, gas, receipts)), reason)) if signedTransactions.length > 1 =>
+        log.debug(s"failure while preparing block because of $reason in transaction with hash ${}")
+        val txIndex = signedTransactions.indexWhere(tx => tx.hash == stx.hash)
+        executePreparedTransactions(signedTransactions.drop(txIndex + 1),
+          worldState, blockHeader, signedTransactionValidator, gas, receipts, signedTransactions.take(txIndex))
```

Comment:
Should we include the `executed` txs in the `executed` parameter for the recursive call here (doing something like `executed++signedTransactions.take(txIndex)`)?
For example if we are executing: `stx1, stx2, stx3, stx4, stx5`, and both `stx2` and `stx4` fail, when `stx4` fails, could it be the case that we are not including `stx1` as executed?